compiler/src/main/java/org/qbicc/type/Primitive.java

```diff
@@ -0,0 +1,86 @@
+package org.qbicc.type;
+
+import java.util.function.Consumer;
+
+public enum Primitive {
+    // Predefine the set of primitive types in Java
+    BOOLEAN("boolean"),
+    BYTE("byte"),
+    SHORT("short"),
+    CHAR("char"),
+    INT("int"),
+    FLOAT("float"),
+    LONG("long"),
+    DOUBLE("double"),
+    VOID("void");
+
+    private final String name;
+    private int typeId;
+    private ValueType type;
```

Comment:
Since these are set late and are shared variables, I think they should be `volatile`.

Reply:
Done